dbt_style_guide.md

```diff
@@ -1,237 +1,415 @@
+  events as (
 
-- *DO NOT OPTIMIZE FOR A SMALLER NUMBER OF LINES OF CODE. NEWLINES ARE CHEAP, BRAIN TIME IS EXPENSIVE*
+      ...
+      where not is_deleted
 
-### Example SQL
-```sql
-with
+  ),
 
-my_data as (
+  -- CTE comments go here
+  events_joined as (
 
-    select * from {{ ref('my_data') }}
+      ...
 
-),
+  )
 
-some_cte as (
+  select * from events_joined
+  ```
 
-    select * from {{ ref('some_cte') }}
+## SQL style guide
+- **DO NOT OPTIMIZE FOR A SMALLER NUMBER OF LINES OF CODE.**  
+  New lines are cheap, brain time is expensive; new lines should be used within  
+  reason to produce code that is easily read.
 
-),
+- Use trailing commas
 
-some_cte_agg as (
+- Indents should be four spaces 
 
-    select
-        id,
-        sum(field_4) as total_field_4,
-        max(field_5) as max_field_5
+- When dealing with long `when` or `where` clauses, predicates should be on a new
+  line and indented.
 
-    from some_cte
-    group by 1
+- Lines of SQL should be no longer than 80 characters and new lines should be used to ensure this.  
+  Example:
+  ```sql
+  sum(
+    case when order_status = 'complete' then order_total end
+  ) as monthly_total,
 
-),
 
-final as (
+  {{ get_windowed_values(
+        strategy='sum',
+        partition='order_id',
+        order_by='created_at',
+        column_list=[
+          'final_cost'
+        ]
+  ) }} as total_final_cost
+  ```
 
-    select [distinct]
-        my_data.field_1,
-        my_data.field_2,
-        my_data.field_3,
+- Field names and function names should all be lowercase
 
-        -- use line breaks to visually separate calculations into blocks
-        case
-            when my_data.cancellation_date is null
-                and my_data.expiration_date is not null
-                then expiration_date
-            when my_data.cancellation_date is null
-                then my_data.start_date + 7
-            else my_data.cancellation_date
-        end as cancellation_date,
+- The `as` keyword should be used when aliasing a field or table
 
-        some_cte_agg.total_field_4,
-        some_cte_agg.max_field_5
+- Fields should be stated before aggregates / window functions
 
-    from my_data
-    left join some_cte_agg  
-        on my_data.id = some_cte_agg.id
-    where my_data.field_1 = 'abc'
-        and (
-            my_data.field_2 = 'def' or
-            my_data.field_2 = 'ghi'
-        )
-    having count(*) > 1
+- Aggregations should be executed as early as possible before joining to another table.
 
-)
+- Ordering and grouping by a number (eg. group by 1, 2) is preferred over listing the column names (see [this rant](https://blog.getdbt.com/write-better-sql-a-defense-of-group-by-1/) for why). Note that if you are grouping by more than a few columns, it may be worth revisiting your model design.

```

Comment:
Can we maybe add a TL;DR for the article here?

Reply:
also a good place to mention dbt_utils.group_by function

Reply:
@SamHarting not adding a TL;DR because many reasons are listed in the article in which I think a TL;DR would be more than a one-liner. We also didn't do that (except for stating the main points) for any of our other links! If you have something you had in mind we can definitely start up a convo about it/change it!

@lbenezra-FA Added the suggestion for the group_by function. [Commit](https://github.com/dbt-labs/corp/pull/63/commits/c18398f700c0bcc8e175863fe5b044912c784a1b)